Structure revamp

bbuman opened the pull request:
Major update to the client.

Added various methods for querying the Meteomatics API.
Added examples that demonstrate how to use the client.
Added documentation for the client.
Added various unit and integration tests.

Commit:
Initial commit

chess-equality committed:
Initial project setup

chess-equality committed:
Start refactor

chess-equality committed:
Added connector components

chess-equality committed:
Modelled ValidDateTime

chess-equality committed:
Remove unnecessary

chess-equality committed:
Use generics TimeZone for offset

chess-equality committed:
Modelled Parameters; added time_list; other refactoring

chess-equality committed:
Modelled Optionals

chess-equality committed:
Modelled Locations; refactoring

chess-equality committed:
Start query time series

chess-equality committed:
Added VDTOffset enum wrapper; other refactoring

chess-equality committed:
Removed LinkedHashSet; other refactoring

chess-equality committed:
Removed #derive Builder

chess-equality committed:
Started CSV response processing

chess-equality committed:
Use thiserror

chess-equality committed:
Start ConnectorResponse

chess-equality committed:
Refactored

chess-equality committed:
Convert to matrix results (ResponseBody)

chess-equality committed:
Handle period date and time step

chess-equality committed:
Minor refactoring

bbuman-meteomatics committed:
Pull apart values and index

This changes the the way the response is parsed into the ResponseBody.
The records and indexes are now separate.

bbuman-meteomatics committed:
modification to iteration over http response

This no longer uses the enumeration of the response records. It now uses the index 'i' to get both the correct response_index as well as the corresponding response_record.

bbuman-meteomatics committed:
Removed the modules optionals and parameters

They are replaced by vectors

bbuman-meteomatics committed:
Fixed a small typo

bbuman-meteomatics committed:
Change locations and change ConnectorResponse

> The locations are now similar to the Python Connector implementation. A Vec<Vec<f32>>
    the elements of the inner vector are combined using "," while the elements of the outer vector are combined using "+".
> The ConnectorResponse:
    This now temporarily writes a http_response.csv file (<- might cause issues with threads?)
    From this CSV file it produces a polars::frame::DataFrame

bbuman-meteomatics committed:
Make use of dotenv to handle API credentials

Place a file ".env" in the project, at the same level of the Cargo.toml (i.e. the root of the project). Then add these lines to the file:
METEOMATICS_USER=username
METEOMATICS_PW=password

bbuman-meteomatics committed:
Remove temporarily file I/O during DF creation

This new version uses a std::io::Cursor as input to polars::io::csv::CsvReader. This avoids having to write the HTTP response to a temporary CSV and then reading this back in using the polars CSVReader.

bbuman-meteomatics committed:
Remove dependency on tempfile

Creating or using temporary files and directories is no longer required.

bbuman-meteomatics committed:
Added PolarsError to ConnectorError

This allows to remove the Box<dyn std::error::Error>

bbuman-meteomatics committed:
Revamping the structure of the library

> Remove module "configuration"
> Move api_client.rs to src/client.rs
> Use lib.rs simply as accessor to the library functionality (--> remove all the tests that previously were in here)
> Create directory "tests" which is the recommended way to test the integration of the library
> Add the tests that previously were in lib.rs to integration_test.rs
> Adapt different files to the new hierarchy

bbuman-meteomatics committed:
Ignore VSCode launch configuration

bbuman-meteomatics committed:
Merge branch 'develop' of github.com:bbuman/rust-connector-api into structure_revamp

bbuman-meteomatics committed:
Change coords to f64 and add unit test

The API allows at most 6 digits after the decimal point. In testing I realized that the f32 data type is not always correctly converted to string literals in the format!() macro. This issue is resolved by using f64 data types for the coordinates.
Also I added a unit test for the coords_to_str() function.

bbuman-meteomatics committed:
add comment to unit test

bbuman-meteomatics committed:
Complete refactor of the API client

Getting rid of many modules and custom structures.
Time functionality:
> Remove valid_date_time module: the refactored functions instead need stard_date, end_date, and duration (identical to the Python connector)
Format functionality:
> Remove format module: this module functionality is replaced with "/csv" in the query, we will then create specific functions for non-csv downloads (identical to the Python connector e.g. query_netcdf()
> Add more unit tests
> Simplify existing unit tests

bbuman-meteomatics committed:
Small refactor

Building the final url is now no longer part of the do_http_get() function

bbuman-meteomatics committed:
Refactor, add Point struct

bbuman-meteomatics committed:
Rename APIError to ReqwestError

The APIError was not really an API specific error but rather a new name for related reqwest::Error errors. As such I rename it to ReqwestError.

bbuman-meteomatics committed:
Add Documentation and minor fix

bbuman-meteomatics committed:
More integration tests

Checking the DataFrames produced by the API by comparing with CSV outputs from the Python API connector results.
Added tests for both postal and latlon queries. All tests pass.

bbuman-meteomatics committed:
Add resolution specifiers to the BBox

bbuman-meteomatics committed:
Add the ability to query for a grid

Uses the BBox struct to define the areal extent of the grid as well as the resolution.

bbuman-meteomatics committed:
Add more tests

Added several more tests to check if the output DataFrame is consistent with the Python connector and in some cases consistent with the CSV directly downloaded from the API.

bbuman-meteomatics committed:
Add test for NetCDF download

Temporarily creates a file for testing of NetCDF download, then deletes it again.

bbuman-meteomatics committed:
Update Documentation

bbuman-meteomatics committed:
Add netcdf download

Also changed some minor things on the query generation

bbuman-meteomatics committed:
Add crate netcdf

This crate is used in development to test the content of the downloaded NetCDF file. The extracted data values are compared against values extracted in Python using the XArray library.

bbuman-meteomatics committed:
Minor fix

bbuman-meteomatics committed:
Move all relevant files into the meteomatics dir

bbuman-meteomatics committed:
Add module util

Util bundles different functions and structs that are used in the client. This makes the client module easier to maintain because it really only defines the main functionality of the client.
All tests still pass.

bbuman-meteomatics committed:
Add test for user features query

bbuman-meteomatics committed:
Add examples directory

This contains examples of how to work with the rust connector

bbuman-meteomatics committed:
Add PNG download and test

Client is now able to download PNG files.
Tests could be more extensive. Currently only tests the shape.

bbuman-meteomatics committed:
Add example for grid queries

bbuman-meteomatics committed:
Add NetCDF example query

bbuman-meteomatics committed:
Ignore test outputs and other (large) files

PNG
CSV
NETCDF

bbuman-meteomatics committed:
Add Query Grid PNG Timeseries function

This function iterates the time series (current date + interval while <= end_date) and calles the query_grid_png function for each step.

bbuman-meteomatics committed:
Add a test for the quer grid png timeseries func

bbuman-meteomatics committed:
Refactor

Implement suggestions based on ```cargo clippy```.

bbuman-meteomatics committed:
Working on documentation

bbuman committed:
Merge branch 'chess-equality:main' into structure_revamp

anicicn84 commented on examples/grid_queries/src/main.rs:
```diff
@@ -0,0 +1,68 @@
+
+    // Do some calculations
+    for col in vec!["t_2m:C", "precip_1h:mm"] {
+        let mean: f64 = df_ts[col].mean().unwrap();
```
I would now use unwraps in a production code. Maybe expect, or handle this to show why the error occurred in the 1st place.

anicicn84 commented on examples/grid_queries/src/main.rs:
```diff
@@ -0,0 +1,68 @@
+
+    // Do some groupby calculations
+    for col in vec!["t_2m:C", "precip_1h:mm"] {
+        let lat_means = df_ts.groupby(["lat"]).unwrap().select(&[col]).mean().unwrap();
```
https://stackoverflow.com/questions/39477684/should-i-avoid-unwrap-in-production-application

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+    let dir: &Path = Path::new(file_name).parent().unwrap();
+    let b: bool = dir.is_dir();
+
+    if !b {
```
if !dir.is_dir, why do you need this temp variable?

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+           Ok(_) => Ok(()),
+           Err(_) => Err(ConnectorError::FileIOError)
+       }
+    } else {
```
Probably you do not need this else, since you match and return Ok or Error, so the code below will anyway not be hit.

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+    // TODO: Convert to simplified version see postal code
+    let mut lat: Vec<f64> = Vec::new();
+    let mut lon: Vec<f64> = Vec::new();
+    for _ in 0..n {
```
You push the same point here?

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+    }
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.extend
+    // https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.get_column_names
+    let df_tmp = df!("lat" => &lat, "lon" => &lon)?;
```
If you push the same point here you can use, maybe:
let df_tmp = df!("lat" => &vec![point.lat; df_in.height()];, "lon" => &vec![point.lon; df_in.height())?; 
Can this work, I mean if we already have a temp vectors of all the same points?

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+        let file_name: String = String::from("tests/netcdfs/my_netcdf.nc");
+        crate::util::create_path(&file_name).await.unwrap();
+        let dir: &Path = Path::new(&file_name).parent().unwrap();
+        let check: bool = dir.is_dir();
```
you do not need this misleading name "check" just  to be temp here, just use the whole expression in assert.
assert!(dir.is_dir(), "Directory does not exist"), you do not need eq for true.

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+        let p2: Point = Point { lat: -52.520551, lon: 13.461804};
+        let coords: Vec<Point> = vec![p1, p2];
+        let coord_str = crate::util::points_to_str(&coords).await;
+        assert_eq!("52.520551,13.461804+-52.520551,13.461804", coord_str);
```
2 magic variables, I would name them at the top and use them for Point struct and assert, otherwise if they change you will have to change them in 2 different places, it is prone to bugs then.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+
+    // Credentials
+    dotenv().ok();
+    let api_key: String = env::var("METEOMATICS_PW").unwrap();
```
Maybe instead of unwrap tell to the user, or runner of these tests that he/she has to have these env variables set, otherwise the error (panic) might be misleading.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .query_time_series(&start_date, &end_date, &interval, &parameters, &coords, &None)
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
Do you need printlns in tests? Better is to assert, than to pollute the console with an unnecessary output.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .query_time_series(&start_date, &end_date, &interval, &parameters, &coords, &None)
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
I would remove printlines from tests since they contribute nothing. In many testing frameworks in other languages outputting to console out is not permitted.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .query_time_series_postal(&start_date, &end_date, &interval, &parameters, &postal1, &None)
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .query_grid_pivoted(&start_date, &parameter, &bbox, &None)
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .query_grid_unpivoted(&start_date, &parameters, &bbox, &None)
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        )
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .query_time_series(&start_date, &end_date, &interval, &parameters, &coords, &None)
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .await
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+           println!("{:?}", df);
+        }
+        Err(ref connector_error) => {
+            println!(">>>>>>>>>> ConnectorError: {:#?}", connector_error);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        }
+        Err(ref connector_error) => {
+            println!(">>>>>>>>>> ConnectorError: {:#?}", connector_error);
+            assert!(result.is_err());
```
You can pass this error you want to println as the 2nd argument to the assert.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+            println!("{:?}", df);
+        }
+        Err(ref connector_error) => {
+            println!(">>>>>>>>>> ConnectorError: {:#?}", connector_error);
```
```suggestion
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        }
+        Err(ref connector_error) => {
+            println!(">>>>>>>>>> ConnectorError: {:#?}", connector_error);
+            assert!(result.is_err());
```
You can pass this string to print as the 2nd arg to assert macro

bbuman replied:
To be more consistent with the other tests I rewrote the first two tests ( ```call_query_time_series_without_options()``` and ```call_query_time_series_with_options()```). They now also use actual response data to compare two DataFrames.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+
+    let ustats = meteomatics_connector.query_user_features().await.unwrap();
+    assert_eq!(env::var("METEOMATICS_USER").unwrap(), ustats.stats.username);
+    assert_eq!(true, ustats.stats.area);
```
```suggestion
    assert!(ustats.stats.area, "Custom message here if not true");
```

bbuman replied:
This is not a good test in itself (no matter if used in assert_eq! or assert!) because it is perfectly valid for ustats.stats.area to be false and then both variants of this test will fail. The only real test where we know the output at runtime is that for the user name. 

Other tests for deserialization are now included in the util.rs module (see ```check_deserialization```).

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+    // Remove the file    
+    let dir: &Path = Path::new(&file_name).parent().unwrap();
+    fs::remove_file(&file_name).unwrap();
+    fs::remove_dir_all(&dir).unwrap();
```
Maybe at the end some assertin that the file is actually removed, like checking for existence and it should return false. 
assert!(!path_exists)

bbuman replied:
Good idea, thanks!

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+
+    // Remove the file    
+    let dir: &Path = Path::new(&file_name).parent().unwrap();
+    fs::remove_file(&file_name).unwrap();
```
assert the file does not exist anymore.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+    // Remove the file    
+    let dir: &Path = Path::new(&file_name).parent().unwrap();
+    fs::remove_file(&file_name).unwrap();
+    fs::remove_dir_all(&dir).unwrap();
```
assert the file and dire do not exist anymore at the end.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+    let interval = Duration::hours(12);
+
+    // Create Parameters
+    let parameter =String::from("t_2m:C");
```
```suggestion
    let parameter = String::from("t_2m:C");
```

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+    };
+
+    // Create file name
+    let file_name: String = String::from("tests/png/my_png.png");
```
You do not need a type here, it can be deduced, but okay. 
Especially since you are calling String::from, quite visible here what it is.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+    assert_eq!(xr_test, temp_f64);
+
+    // Check another value: ds_rust["t_2m"].data[2,2,2]
+    let xr_test: f64 = 5.269172668457031;
```
Variable name does not tell much.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+
+    // Make some tests
+    let file = netcdf::open(&file_name).unwrap();
+    let var = &file.variable("t_2m").expect("Could not find variable 't_2m");
```
var is too generic name of the variable.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+        .unwrap();
+    println!("Rust result: {:?}", df_q);
+    println!("Python result: {:?}", df_s);
+    assert!(df_s.frame_equal(&df_q));
```
Maybe custom message here useful to the person running the tests if they are not equal, why so,

bbuman replied:
It is difficult to anticipate why the two DataFrames might turn out to be unequal.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+
+    match result {
+        Ok(ref df) => {
+            println!("{:?}", df);
```
println in tests does nothing, maybe assert!(true) or something.

bbuman replied:
Removed the whole match statement.

anicicn84 commented on meteomatics/tests/integration_test.rs:
```diff
@@ -0,0 +1,785 @@
+            println!("{:?}", df);
+        }
+        Err(ref connector_error) => {
+            println!(">>>>>>>>>> ConnectorError: {:#?}", connector_error);
```
Maybe here assert!(false) or something.

bbuman replied:
removed

anicicn84 commented on meteomatics/src/util.rs:
```diff
@@ -0,0 +1,399 @@
+        // TODO: Rename variable
+        let file = std::io::Cursor::new(&body);
+        use polars::prelude::*; 
+        let df1 = polars::io::csv::CsvReader::new(file)
```
df1 is too generic, maybe first_data_frame, or something more meaningful. It is hard to understand what it is this way.

anicicn84 commented on examples/point_queries/src/main.rs:
```diff
@@ -0,0 +1,58 @@
+    // Print the query result
+    println!("{:?}", df_ts);
+
+    // Do some calculations
```
Comment does not add to a value.

anicicn84 commented on examples/netcdf_queries/src/main.rs:
```diff
@@ -0,0 +1,63 @@
+    let param1: String = String::from("t_2m:C");
+
+    // Optionals
+    let opt1: String = String::from("model=mix");
```
⏫ 
These 2 variable names are terrible :) 
They should be more descriptive about their purpose and what do they do, from the type we can see they are optionals, so we have 2 informations about them being optionals, but 0 about what are they actually underneath the hood.
⏬

anicicn84 commented on examples/netcdf_queries/src/main.rs:
```diff
@@ -0,0 +1,63 @@
+
+    // Look at the NetCDF
+    let file = netcdf::open(&file_name).unwrap();
+    let var = &file.variable("t_2m").expect("Could not find variable 't_2m");
```
var, file, too general names for the variables. 
Code is written to be read by other developers in the easiest way possible, just having a `file` variable, or `var` does not add to a value of the code.

anicicn84 commented on examples/grid_queries/src/main.rs:
```diff
@@ -0,0 +1,68 @@
+    let params: Vec<String> = vec![param1, param2];
+
+    // Optionals
+    let opt1: String = String::from("model=mix");
```
Please do change these 2 variable names

bbuman-meteomatics committed:
Remove "NA"

bbuman-meteomatics committed:
Implement Lighnting Query

bbuman-meteomatics committed:
remove comment

bbuman-meteomatics committed:
Rename columns of DataFrame for lightning query

bbuman-meteomatics committed:
Fix compile errors in example

bbuman-meteomatics committed:
Adapt to new output of user_stats_json query

bbuman-meteomatics committed:
Add route query using Points

bbuman-meteomatics committed:
Add tests for lightning and route query

bbuman-meteomatics committed:
Add route query for postal codes

bbuman-meteomatics committed:
Add test for route query using postal codes

bbuman-meteomatics committed:
Refactor

using suggestions from clippy

bbuman-meteomatics committed:
Better tests for two tests

call_query_time_series_without_options() and call_query_time_series_with_options() now also make use of sample data converted to a sample dataframe to check if the query returns the same result.

bbuman-meteomatics committed:
Add unit test for the deserialization

This unit test checks that serde_json can correctly deserialize the HTTP response JSON for the user_stats_json query.

bbuman-meteomatics committed:
Remove assertion

This assertion is not always valid and may fail erronously.

bbuman-meteomatics committed:
Add new assertions

These assertions check if files and directories were removed as intended.

bbuman-meteomatics committed:
Refactor variable names

bbuman-meteomatics committed:
Add documentation, use dotenv

bbuman-meteomatics committed:
Add documentation, use dotenv

bbuman-meteomatics committed:
Add documentation, use dotenv

bbuman-meteomatics committed:
Add more documentation

bbuman-meteomatics committed:
Adding examples to the public functions

bbuman-meteomatics committed:
Remove type hints

bbuman-meteomatics committed:
Add reexport of Point and BBox

This simplifies the access to Point and BBox for users of the library.

bbuman-meteomatics committed:
Use Point and BBox from client instead of location

bbuman-meteomatics committed:
Fix examples for Point and BBox

cargo test -> ok

bbuman-meteomatics committed:
Add documentation, allow compilation test

cargo test => ok

bbuman-meteomatics committed:
Fix bugs in examples

The examples now compile (cargo test => ok)

bbuman-meteomatics committed:
Add description change minor version

Incremented minor version to 2 (0.2.0)

bbuman-meteomatics committed:
Minor change to the example

Use TimeZone instead of prelude

bbuman committed:
Create ci.yml

bbuman-meteomatics committed:
Merge remote-tracking branch 'origin/main' into structure_revamp

bbuman committed:
Create rust.yml

bbuman committed:
Update rust.yml

Add working-directory

bbuman-meteomatics committed:
Merge remote-tracking branch 'origin/main' into structure_revamp

bbuman-meteomatics committed:
Remove workflow

bbuman-meteomatics committed:
Change workflow working directory

bbuman-meteomatics committed:
Change working directory

bbuman-meteomatics committed:
change working directory

bbuman-meteomatics committed:
Deactivate workflow due to missing HDF5

Error in workflow:  "failed to run custom build command for `hdf5-sys v0.8.1`"

bbuman-meteomatics committed:
Enable workflow

The workflow can be disabled on GitHub instead of commenting out the whole yml file.

bbuman-meteomatics committed:
Try manual install of HDF5 to fix error

bbuman-meteomatics committed:
Add option to manually trigger workflow

bbuman-meteomatics committed:
Also install build essential

Attempt to fix "error: linking with `cc` failed: exit status: 1"

bbuman-meteomatics committed:
Only test lib and doc

bbuman-meteomatics committed:
Fix typo

bbuman-meteomatics committed:
add working directory to library test

bbuman-meteomatics committed:
Query mockup server for the status

The mockup server allows the continuous integration to run the "client_fires_get_request" test.

bbuman committed:
Update README.md

Adding workflow badge.

bbuman-meteomatics committed:
Remove the mockup-server query

Apparently we can also call the status endpoint on the actual meteomatics api without credentials. So the mockup-server is not required for this test. It might however come in handy for other tests down the line (e.g. in the integration tests).

bbuman-meteomatics committed:
Change edition to 2021

bbuman-meteomatics committed:
Major refactor to use new struct "TimeSeries"

This allows to comply to the clippy warning of too many parameters. Start, end and timedelta are now integrated into the TimeSeries struct.
> Adapted examples
> Adapted documentation
> Adapted tests

bbuman-meteomatics committed:
Refactor and add argument descriptions to docs

bbuman-meteomatics committed:
Adapt to refactored client

bbuman-meteomatics committed:
Move examples into the connector

Now we can use "cargo run --example point_query" to run this example. Dependencies are automatically the same as for the library.

bbuman-meteomatics committed:
Rename files

bbuman-meteomatics committed:
Add examples, more documentation for the examples

bbuman-meteomatics committed:
Fix references

bbuman-meteomatics committed:
Update with examples

bbuman-meteomatics committed:
fix variable naming mistake

bbuman-meteomatics committed:
Add more examples for different query types

Based on https://github.com/meteomatics/python-connector-api/tree/master/examples

bbuman-meteomatics committed:
Remove dependency on the netcdf library

Currently thdisabled because [`netcdf`] depends on HDF5 being installed on the target operating system. Please install HDF5 on your platform before you run these tests. Then also include netcdf = "0.7.0" in the dependencies (uncomment the respective line in the Cargo.toml under developent dependencies).

bbuman-meteomatics committed:
Add fn main(){unimplemented!()} to empty examples.

This fixes compilation issues with empty rust files.

bbuman-meteomatics committed:
Fix tiny typo

bbuman-meteomatics committed:
use unimplemented macro

bbuman-meteomatics committed:
Update Readme

meteomatics committed:
Merge branch 'main' into structure_revamp
